k8s/chart/templates/deployment.yaml

```
@@ -0,0 +1,134 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "sdp-operator.fullname" . }}
+  labels:
+  {{- include "sdp-operator.labels" . | nindent 4 }}
+spec:
+  replicas: 1q
+  selector:
+    matchLabels:
+      {{- include "sdp-operator.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "sdp-operator.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.sdp.operator.image.pullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      serviceAccountName: {{ include "sdp-operator.serviceAccountName" . }}
+      containers:
+        - name: {{ .Chart.Name }}
+          env:
+            - name: APPGATE_OPERATOR_NAMESPACE
+              value: {{ .Release.Namespace }}
+            - name: APPGATE_OPERATOR_USER
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "sdp-operator.secret" . }}
+                  key: appgate-operator-user
+            - name: APPGATE_OPERATOR_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "sdp-operator.secret" . }}
+                  key: appgate-operator-password
+            - name: APPGATE_OPERATOR_HOST
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ include "sdp-operator.config" . }}
+                  key: appgate-operator-host
```

Can we pass the values directly here instead of passing it through a configmap?
```suggestion
            - name: APPGATE_OPERATOR_HOST
              value: {{ required ".values.sdp.operator.host is  required" .Values.sdp.operator.host }}
```

---

By the way, helm has a bunch of bult-in functions to assist validating values: https://helm.sh/docs/chart_template_guide/function_list/